### PR TITLE
Backpropagation implementation of  mergemax, mergeadd and mergeavg ops

### PR DESCRIPTION
**Pull request by olegsemeniv**
## What changes were proposed in this pull request?

Implemented backpropagation for:
- mergemax;
- mergeadd;
- mergeavg.

Split implementation of transforms operations into separate files for compilation speed up.

## How was this patch tested?

test cases added, manually checked with gradCheck() utility. 

## Quick checklist

The following checklist helps ensure your PR is complete:

- [x] Eclipse Contributor Agreement signed, and signed commits - see [IP Requirements](https://deeplearning4j.org/eclipse-contributors) page for details
- [x] Reviewed the [Contributing Guidelines](https://github.com/eclipse/deeplearning4j/blob/master/CONTRIBUTING.md) and followed the steps within.
- [x] Created tests for any significant new code additions.
- [x] Relevant tests for your changes are passing.

### CODE REVIEW

**Review by raver119 — DISMISSED** [review]

File: libnd4j/include/ops/declarable/helpers/cpu/transforms_merge.cpp [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+    
+    const Nd4jLong numArgs = outArrs.size();
+    // last array is gradient
+    auto gradient = inArrs[numArgs]->bufferAsT<T>();
+    auto length = inArrs[numArgs]->lengthOf();
+    
+    bool bSameOrderAndEws1 = true;
+
+    auto ordering = inArrs[numArgs]->ordering();
+    for (auto v : inArrs) {
```
Comment (shyrma) [inline libnd4j/include/ops/declarable/helpers/cpu/transforms_merge.cpp]: let check here the possibility to use same offsets in slower loop, I mean
... &= shape::haveSameShapeAndStrides

**Review by shyrma — COMMENTED** [review]
also I would suggest to change renaming procedure from transform_nameOp to nameOp

**Comment by raver119** [comment]
> also I would suggest to change renaming procedure from transform_nameOp to nameOp

Yep, that will be better. I'm going to rearrange helper files in the same order/folders as op implementations eventually, for better navigation.

File: libnd4j/include/ops/declarable/helpers/cpu/transforms_merge.cpp (hunk 1 above)
Comment (olegsemeniv) [reply libnd4j/include/ops/declarable/helpers/cpu/transforms_merge.cpp]: corrections added per request. Files names changed.

**Pull request merged**